tests/integration/test_unpack.py

```diff
@@ -0,0 +1,111 @@
+from guppylang.decorator import guppy
+from guppylang.module import GuppyModule
+from guppylang.std.builtins import array, owned
+
+from guppylang.std.quantum import qubit
+
+
+def test_unpack_array(validate):
+    module = GuppyModule("test")
+    module.load(qubit)
+
+    @guppy(module)
+    def main(qs: array[qubit, 3] @ owned) -> tuple[qubit, qubit, qubit]:
+        q1, q2, q3 = qs
+        return q1, q2, q3
+
+    # validate(module.compile())
+    module.check()
+
+
+def test_unpack_starred(validate):
+    module = GuppyModule("test")
+    module.load(qubit)
+
+    @guppy(module)
+    def main(
+        qs: array[qubit, 10] @ owned,
+    ) -> tuple[qubit, qubit, qubit, qubit, qubit, qubit, array[qubit, 4]]:
+        q1, q2, *qs, q3 = qs
+        [q4, *qs] = qs
+        *qs, q5, q6 = qs
+        [*qs] = qs
+        return q1, q2, q3, q4, q5, q6, qs
```

Comment:
q: Is there an issue for supporting the _rhs_ case of unpacking, i.e. writing something like this?
```py
return q1, q2, q3, q4, q5, q6, *qs
```

(currently it panics with a `NotImplementedError`)

Reply:
Created #690 👍

Reply:
Also, we shouldn't be throwing `NotImplementedError` 😅 See #691